### PR TITLE
Rename `expirationTime`/`lifetime` to what it really is: an idle timeout

### LINKED ISSUE #553: Sessions may never expire: make Renewal Timeout opt-in, implement Absolute Timeout
There are 3 [Automatic Session Expiration](https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#automatic-session-expiration) mechanism this library should provide:

1. [Idle Timeout](https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#idle-timeout): after a period of inactivity, a session must expire. That's already implemented
2. [Renewal Timeout](https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#renewal-timeout): if you keep working actively, your session gets renewed. That's already implemented, but be completely disabled only partially
3. [Absolute Timeout](https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#absolute-timeout): no matter what, the session must expire after a defined time window. That's missing

## Issue 1/3: Renewal Timeout is a hidden behaviour

Currently both [`README.md`](https://github.com/psr7-sessions/storageless/blob/8.13.x/README.md) and [`docs/configuration.md`](https://github.com/psr7-sessions/storageless/blob/8.13.x/docs/configuration.md) lack of a proper explanation of what's going on under the hood, which is that by default the session is **potentially infinite** if the user keeps interacting with the application and the configured `$expirationTime` is greater than `$refreshTime`, and that's highly likely given that the default `$refreshTime` is 60 seconds.

This is even worse at a communication level considering that it deeply differs from the default `ext-session` behaviour; I bet most consumer of this library expect just replacement of code, not of timeout mechanisms.

*Solution*: write a better documentation

## Issue 2/3: Renewal Timeout should be opt-in

Renewal Timeout mechanism it's not recommended to be a default.
Re-authentication should occur if not explicitly configured otherwise.

I'm using the undocumented workaround of setting `$refreshTime = 1 + $expirationTime` to disable it, if anyone wonders how to opt-out in current major release.

*Solution*: disable the default Renewal Timeout, make it opt-in. https://github.com/psr7-sessions/storageless/labels/BC%20break

## Issue 3/3: Absolute Timeout is unachievable

Even when the Refresh Timeout is disabled with the `$refreshTime = 1 + $expirationTime` trick, the session gets renewed if the content changes, and the Timeout shifts afar again because the expiration time always starts from `$this->clock->now()`:

https://github.com/psr7-sessions/storageless/blob/fb7ab9550ad0181de0432c7749c5f261f3fceaf7/src/Storageless/Http/SessionMiddleware.php#L201-L203

https://github.com/psr7-sessions/storageless/blob/fb7ab9550ad0181de0432c7749c5f261f3fceaf7/src/Storageless/Http/SessionMiddleware.php#L224-L225

*Solution*: when Renewal Timeout is disabled, inherit `$expiresAt` value from current Session token. https://github.com/psr7-sessions/storageless/labels/BC%20break

<hr>

Ping @malukenho and @lcobucci for feedbacks, you've worked on this in https://github.com/psr7-sessions/storageless/commit/89719b36afc8321dd143efe074f81aef337ae457 and https://github.com/psr7-sessions/storageless/pull/14

I'll propose a PR once you've reviewed this issue.

### PR DESCRIPTION
**Pull request by Slamdunk**
Fix https://github.com/psr7-sessions/storageless/issues/553 issue 1

### CODE REVIEW

File: src/Storageless/Http/SessionMiddleware.php [hunk 1]
```diff
@@ -61,7 +61,7 @@ final class SessionMiddleware implements MiddlewareInterface
     public function __construct(
         Configuration $configuration,
         SetCookie $defaultCookie,
-        private int $expirationTime,
+        private int $idleTimeout,
```
Comment (Slamdunk) [inline src/Storageless/Http/SessionMiddleware.php:64]: Schrödinger's https://github.com/psr7-sessions/storageless/labels/BC%20break: it's backward incompatible only if you care about [named parameters](https://wiki.php.net/rfc/named_params#vote)

**Review by Ocramius — APPROVED** [review]
This is a very acceptable improvement, IMO 👍

File: src/Storageless/Http/SessionMiddleware.php (hunk 1 above)
Comment (Ocramius) [reply src/Storageless/Http/SessionMiddleware.php:64]: True, although IMO not worth pursuing this 👍

Comment (Ocramius) [reply src/Storageless/Http/SessionMiddleware.php:64]: Also, TIL that you can link labels, awesome!

**Comment by Ocramius** [comment]
Thanks @Slamdunk!

**Pull request merged**